src/vmm_mad/remotes/lib/vcenter_driver/host.rb

```diff
@@ -94,6 +96,177 @@ def get_resource_pool_list(rp = @item.resourcePool, parent_prefix = "", rp_array
         rp_array
     end
 
+    def get_nsx
+        nsx_info = ""
+        nsx_obj = {}
+        extensionList = []
+        extensionList = @vi_client.vim.serviceContent.extensionManager.extensionList
+        extensionList.each do |extList|
+            if extList.key == "com.vmware.vShieldManager"
+              nsx_obj['type'] = "NSX-V"
+              nsx_obj['version'] = extList.version
+              nsx_obj['label'] = extList.description.label
+                urlFull = extList.client[0].url
+                urlSplit = urlFull.split("/")
+                # protocol = "https://"
+                protocol = urlSplit[0] + "//"
+                # ipPort = ip:port
+                ipPort = urlSplit[2]
+                nsx_obj['url'] = protocol + ipPort
+            elsif extList.key == "com.vmware.nsx.management.nsxt"
+              nsx_obj['type'] = "NSX-T"
+              nsx_obj['version'] = extList.version
```

Comment:
nsx_obj['version'] abd nsx_obj['label'] looks like they can be taken out of the if